controllers/storagecluster/generate.go

```diff
@@ -45,10 +45,25 @@ func generateNameForCephBlockPool(initData *ocsv1.StorageCluster) string {
 }
 
 func generateDeviceClass(initData *ocsv1.StorageCluster) string {
-	if initData.Spec.ManagedResources.CephNonResilientPools.Enable {
-		return "replicated"
+	if len(foundDeviceClasses) == 0 {
+		if initData.Spec.ManagedResources.CephNonResilientPools.Enable {
+			return "replicated"
+		}
+		// if Non-Resilient pools are not enabled then keep the device class blank
+		return ""
+	} else if len(foundDeviceClasses) == 1 {
+		return foundDeviceClasses[0].Name
+	} else if len(foundDeviceClasses) == len(replica1DeviceClasses)+1 {
+		deviceClassMap := make(map[string]bool)
```

Comment:
This naming will help for readability
```suggestion
		replica1DeviceClassMap := make(map[string]bool)
```